query/common/hll.go

```diff
@@ -688,11 +688,36 @@ func (hll *HLL) Encode() []byte {
 	if len(hll.DenseData) != 0 {
 		return hll.DenseData
 	}
-	data := make([]byte, 3*len(hll.SparseData))
+	return hll.encodeSparse(false)
+}
+
+// EncodeBinary converts HLL to binary format
+// aligns to 4 bytes for sparse hll
+// used to build response for application/hll queries from HLL struct
+func (hll *HLL) EncodeBinary() []byte {
+	if len(hll.DenseData) != 0 {
+		return hll.DenseData
+	}
+	return hll.encodeSparse(true)
+}
+
+func (hll *HLL) encodeSparse(padding bool) []byte {
+	recordValueBytes := 3
+	if padding {
+		recordValueBytes = 4
+	}
+	data := make([]byte, (recordValueBytes)*len(hll.SparseData))
 	for i, register := range hll.SparseData {
-		data[i*3] = byte(register.Index & 0xff)
-		data[i*3+1] = byte(register.Index >> 8)
-		data[i*3+2] = register.Rho
+		if padding {
+			data[i*recordValueBytes] = byte(0)
```

Comment:
actually better to do *(*uint32)(unsafe.Pointer(&data[i*recordValueBytes])) = uint32(register.Rho << 16 | register.Index), also could you add a test using readHLL to read the byte value back to HLL?

Reply:
register.Rho << 16 won't really work.. 
even if it works, `|` between byte and uint16 wont' work..
it will need a lot of padding 0s and cast to uint32 for both `Rho` and `Index`, won't be cleaning than current way

Reply:
I can add some comment to make it readable

Reply:
nvm, looks like I just need to cast the 2 parts separately..